### ISSUE TITLE
Potentional bug in StreamRequestHandler

### ISSUE THREAD

**Issue by bartatamas** [vaadin/flow#6251]
In my application I can't map Vaadin servlet to /*, instead I map it to /vaadin/*.
It seems it works, but the Upload component displays error it sends request to /VAADIN/dynamic/...
I have to add the /VAADIN/* mapping to the Vaadin servlet to handle these type of requests.

The StreamRequestHandler should handle this request but it won't because pathInfo doesn't starts with "VAADIN/dynamic/resource/"

The pathInfo is the request URL after servlet path, so it's real value is /dynamic/resource/... if I map Vaadin servlet to /VAADIN/*

So my opinion is that this versions only works for /* mapping and getPathInfo() should be changed to getSerlvetPath() + getPathInfo() to be able to work with /VAADIN/* mapping too.

Vaadin version: 14.0.0.rc9
Flow version: flow-server-2.0.7
Java version: 8
OS version: Ubuntu 18.04

**Comment by caalador** [comment]
`StreamResourceRegistry::getURI` should probably take into account the context which it right now doesn't seem to do.

**Comment by bartatamas** [comment]
> `StreamResourceRegistry::getURI` should probably take into account the context which it right now doesn't seem to do.

In this test case, the problem is not about servlet context (maybe it is a problem too), but servlet path.

**Comment by caalador** [comment]
Yes but from the look into the code Upload gets the knowledge of call `/VAADIN/dynamic/...` when it should get `vaadin/VAADIN/dynamic/...` not 100 on if it should be fixed on the server side or the component side.

**Comment by OlliTietavainenVaadin** [comment]
Does the problem occur only with the mapping ` /vaadin/` and not e.g. `/hello/`?

**Comment by bartatamas** [comment]
I guess this problem occurs for every mapping other than /, so it should happen with /hello/ too
